private/protocol/json/jsonutil/unmarshal.go

```diff
@@ -211,3 +227,33 @@ func unmarshalScalar(value reflect.Value, data interface{}, tag reflect.StructTa
 	}
 	return nil
 }
+
+func timeFromUnixString(d string) (time.Time, error) {
+	var secs, nsecs int64
+	if dot := strings.Index(d, "."); dot != -1 {
+		var err error
+		secs, err = strconv.ParseInt(d[:dot], 10, 64)
```

Comment:
Another option may be to accept the floating point precision error and allow for overriding if people require more precise values. I think for most cases people will not care too much about the minor precision issues. 

@jasdel, @glasser  - What do you think?